src/resolvers/default.rs

```diff
@@ -140,6 +147,57 @@ impl Dh for Dh25519 {
     }
 }
 
+impl Cipher for CipherAesGcm {
+    fn name(&self) -> &'static str {
+        "AESGCM"
+    }
+
+    fn set(&mut self, key: &[u8]) {
+        copy_slices!(key, &mut self.key)
+    }
+
+    fn encrypt(&self, nonce: u64, authtext: &[u8], plaintext: &[u8], mut out: &mut[u8]) -> usize {
+        let aead = aes_gcm::Aes256Gcm::new(self.key.into());
+
+        let mut nonce_bytes = [0u8; 12];
+        copy_slices!(&nonce.to_le_bytes(), &mut nonce_bytes[4..]);
+
+        let payload = Payload {
+            msg: plaintext,
+            aad: authtext,
+        };
+        
+        match aead.encrypt(&nonce_bytes.into(), payload) {
+            Ok(ciphertext) => {
+                out.write_all(&ciphertext).unwrap();
+                ciphertext.len() as usize
+            },
+            // Bail out if we fail to encrypt the payload
+            Err(e) => panic!("Encryption failed! {:?}", e)
+        } 
+    }
+
+    fn decrypt(&self, nonce: u64, authtext: &[u8], ciphertext: &[u8], mut out: &mut[u8]) -> Result<usize, ()> {
+        let aead = aes_gcm::Aes256Gcm::new(self.key.into());
+
+        let mut nonce_bytes = [0u8; 12];
+        copy_slices!(&nonce.to_le_bytes(), &mut nonce_bytes[4..]);
+
+        let payload = Payload {
+            msg: ciphertext,
+            aad: authtext
+        };
+
+        match aead.decrypt(&nonce_bytes.into(), payload) {
+            Ok(plaintext) => {
+                out.write_all(&plaintext).unwrap();
+                Ok(plaintext.len())
+            }
+            Err(_) => Err(()),
```

Comment:
similar to above, can be accomplished with `map_err()` instead of a match statement

Reply:
Thanks for the feedback, fixed.